comdb2/cdb2.py

```diff
@@ -359,7 +381,7 @@ def row_factory(
     def execute(
         self,
         sql: str | bytes,
-        parameters: Mapping[str, ParameterValue] | None = None,
+        parameters: Mapping[str, ParameterValue] | Sequence[ParameterValue] | None = None,
```

Comment:
Does this have to be a sequence? It looks to me like this could be any iterator, right? In general I prefer permissive inputs and restrictive outputs, where the permissiveness makes sense. I'd say that a generator could make sense to use here, right? I mean, it would have to yield the appropriate number of items, but we don't need to know that number ahead of continuing them, do we?

Reply:
I'm honestly not entirely sure that this makes sense to do as a user. I guess the main question was whether you considered it.

Reply:
This is what [the PEP](https://peps.python.org/pep-0249/#id20) says:

> [.execute](https://peps.python.org/pep-0249/#id20)(operation [, parameters])
> 
> Prepare and execute a database operation (query or command).
> Parameters may be provided as sequence or mapping and will be bound to variables in the > operation. Variables are specified in a database-specific notation (see the module’s [paramstyle](https://peps.python.org/pep-0249/#paramstyle) attribute for details). [[5]](https://peps.python.org/pep-0249/#id53)

The spec says a mapping or a sequence, so that's what we've got here. We could maybe make arbitrary iterables work, but only by collecting them into a sequence ourselves, essentially. We need to know the length before sending the query out over the wire.